### DISCUSSION TITLE
observing intermittent drop in metrics data

### DISCUSSION THREAD

**Discussion by mdvaliyullah** [grafana/mimir#4069] [Help and support]
Running mimir in monolithic mode, observing intermittent drop in metrics data and could see below error in logs. Please advise
```
level=error ts=2023-01-24T06:01:52.550136356Z caller=retry.go:78 user=fake msg="error processing request" try=0 err="context canceled"
level=error ts=2023-01-24T06:01:52.550573358Z caller=frontend_processor.go:88 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Canceled desc = context canceled"
level=error ts=2023-01-24T06:01:52.550713059Z caller=frontend_processor.go:168 msg="error processing requests" err=EOF
```

**Comment by pracucci** [comment]
The logs you shared are related to canceled queries (e.g. when you cancel a query from Grafana UI).


> observing intermittent drop in metrics data

What do you mean by "intermittent drop in metrics data"? Can you share more information about it, please?

**Reply by mdvaliyullah** [reply]
further analysed the logs, the error was not on the mimir side, it was on mimir-proxy.

`ServeHTTP level=error msg=\"failed to push metric data\" err=\"failed writing metrics: remote write API returned HTTP status 503 Service Unavailable: upstream connect error or disconnect/reset before headers. reset reason: connection termination`

**Reply by pracucci** [reply]
I would recommend to install [Mimir dashboards](https://grafana.com/docs/mimir/latest/operators-guide/monitor-grafana-mimir/) and then look at the Mimir cluster health starting from "Overview" and "Writes" dashboard. Distributor and ingester logs could also provide you more details about the actual error which is occurring.
